File: code/web/back-end/routes/travelItem.js
var express = require('express');
var router = express.Router();
var Database = require('../public/javascripts/database.js');
var database = new Database();

var target_global = 'travelItem';

router.get('/query', function (req, res) {
    var user = { tableName : target_global, request : 'query', condition : 'id = ?', values : [req.query.id ]};
    database.query(user, res);
});

router.get('/queryAll', function (req, res) {
    var user = { tableName : target_global, request : 'queryAll', condition : 'travelId = ?', values : [req.query.travelId ]};
    database.queryAll(user, res);
});

router.get('/remove', function (req, res) {
    var user = { id: req.query.id, tableName : target_global, request : 'remove'};
    database.remove(user, res);
});

router.get('/update', function (req, res) {
    var user = { id: req.query.id, tableName : target_global, request : 'update',
        condition : 'travelId = ?, label = ?, time = ?, locationLat = ?, locationLng = ?, like = ?, text = ?, media = ? where id = ?',
        values : [req.query.travelId, req.query.label, req.query.time, req.query.locationLat, req.query.locationLng, req.query.like, req.query.text, req.query.media ]};
    database.update(user, res);
});

router.get('/add', function (req, res) {
    var user = { tableName : target_global, request : 'add',
        condition : '(travelId, label, time, locationLat, locationLng, like, text, media, id) values (?, ?, ?, ?, ?, ?, ?, ?, ?)',
        values : [req.query.travelId, req.query.label, req.query.time, req.query.locationLat, req.query.locationLng, req.query.like, req.query.text, req.query.media ]};
    database.insert(user, res);
});

database.db.run("CREATE TABLE IF NOT EXISTS " + target_global +
                "(id INTEGER PRIMARY KEY NOT NULL, travelId INTEGER," +
                "label TEXT, time TEXT, locationLat REAL, locationLng REAL, like INTEGER," +
                "text TEXT, media TEXT)", function (error) {
        if (error) {
            console.log('create '+ target_global +' table failed: ' + error.toString());
        }
});

module.exports = router;

File: code/android/TravelMap/app/src/main/java/com/jiangzuomeng/modals/TravelItem.java
package com.jiangzuomeng.modals;

import android.content.ContentValues;
import android.net.Uri;
import android.text.method.HideReturnsTransformationMethod;

import java.lang.annotation.Retention;
import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by ekuri-PC on 2015/11/21.
 */
public class TravelItem extends ManLvTuNetworkDataType implements ManLvTuSQLDataType {
    public static final String TRAVEL_ITEM_TABLE_NAME = "travelItem";
    public int id;
    public int travelId;
    public String label;
    public String time;
    public double locationLng;
    public double locationLat;
    public int like;
    public String text;
    public String media;

    public TravelItem() {

    }

    @Override
    public ContentValues makeSQLContentValues() {
        ContentValues contentValues = new ContentValues();
        contentValues.put("travelId", this.travelId);
        contentValues.put("label", this.label);
        contentValues.put("time", this.time);
        contentValues.put("locationLng", this.locationLng);
        contentValues.put("locationLat", this.locationLat);
        contentValues.put("like", this.like);
        contentValues.put("text", this.text);
        contentValues.put("media", this.media);
        return contentValues;
    }

    public static String makeCreateTableSQLString() {
        return "CREATE TABLE IF NOT EXISTS " + TRAVEL_ITEM_TABLE_NAME +
                "(id INTEGER PRIMARY KEY AUTOINCREMENT, travelId INTEGER," +
                "label TEXT, time TEXT, locationLat REAL, locationLng REAL, like INTEGER," +
                "text TEXT, media TEXT)";
    }

    public String makeQueryByTravelItemIdSQLString() {
        return null;
    }

    public String makeQueryByTravelIdSQLString() {
        return null;
    }

    @Override
    public URL getAddUrl() throws MalformedURLException {
        Uri.Builder builder = new Uri.Builder();
        builder.scheme(StaticStrings.HTTP)
                .encodedAuthority(StaticStrings.host)
                .appendPath(StaticStrings.TRAVEL_ITEM)
                .appendPath(StaticStrings.ADD)
                .appendQueryParameter(StaticStrings.TRAVEL_ID, Integer.toString(travelId))
                .appendQueryParameter(StaticStrings.LABEL, label)
                .appendQueryParameter(StaticStrings.TIME, time)
                .appendQueryParameter(StaticStrings.LOCATION_LAT, Double.toString(locationLat))
                .appendQueryParameter(StaticStrings.LOCATION_LNG, Double.toString(locationLng))
                .appendQueryParameter(StaticStrings.LIKE, Integer.toString(like))
                .appendQueryParameter(StaticStrings.TEXT, text)
                .appendQueryParameter(StaticStrings.MEDIA, media);
        URL url = new URL(builder.build().toString());
        return url;
    }

    @Override
    public URL getQueryUrl() throws MalformedURLException {
        Uri.Builder builder = new Uri.Builder();
        builder.scheme(StaticStrings.HTTP)
                .encodedAuthority(StaticStrings.host)
                .appendPath(StaticStrings.TRAVEL_ITEM).appendPath(StaticStrings.QUERY)
                .appendQueryParameter(StaticStrings.ID, Integer.toString(id));
        URL url = new URL(builder.build().toString());
        return url;
    }

    @Override
    public URL getUpdateUrl() throws MalformedURLException {
        Uri.Builder builder = new Uri.Builder();
        builder.scheme(StaticStrings.HTTP)
                .encodedAuthority(StaticStrings.host)
                .appendPath(StaticStrings.TRAVEL_ITEM).appendPath(StaticStrings.UPDATE)
                .appendQueryParameter(StaticStrings.ID, Integer.toString(id))
                .appendQueryParameter(StaticStrings.TRAVEL_ID, Integer.toString(travelId))
                .appendQueryParameter(StaticStrings.LABEL, label)
                .appendQueryParameter(StaticStrings.TIME, time)
                .appendQueryParameter(StaticStrings.LOCATION_LAT, Double.toString(locationLat))
                .appendQueryParameter(StaticStrings.LOCATION_LNG, Double.toString(locationLng))
                .appendQueryParameter(StaticStrings.LIKE, Integer.toString(like))
                .appendQueryParameter(StaticStrings.TEXT, text)
                .appendQueryParameter(StaticStrings.MEDIA, media);
        URL url = new URL(builder.build().toString());
        return url;
    }

    @Override
    public URL getRemoveUrl() throws MalformedURLException {
        Uri.Builder builder = new Uri.Builder();
        builder.scheme(StaticStrings.HTTP)
                .encodedAuthority(StaticStrings.host)
                .appendPath(StaticStrings.TRAVEL_ITEM).appendPath(StaticStrings.REMOVE)
                .appendQueryParameter(StaticStrings.ID, Integer.toString(id));
        URL url = new URL(builder.build().toString());
        return url;
    }

    @Override
    public URL getQueryAllUrl() throws MalformedURLException {
        Uri.Builder builder = new Uri.Builder();
        builder.scheme(StaticStrings.HTTP)
                .encodedAuthority(StaticStrings.host)
                .appendPath(StaticStrings.TRAVEL_ITEM).appendPath(StaticStrings.QUERY_ALL)
                .appendQueryParameter(StaticStrings.TRAVEL_ID, Integer.toString(travelId));
        URL url = new URL(builder.build().toString());
        return url;
    }
}
